src/lib.rs

```diff
@@ -152,43 +199,44 @@ impl LinkDrop {
         );
         let creation_succeeded = is_promise_success();
         if !creation_succeeded {
-            // In case of failure, send funds back.
+            // In case of failure, send attached deposit back.
             Promise::new(predecessor_account_id).transfer(amount.into());
         }
         creation_succeeded
     }
 
     /// Callback after execution `create_account_and_claim`.
-    pub fn on_account_created_and_claimed(&mut self, amount: U128) -> bool {
+    pub fn update_nft_storage(&mut self, public_key: PublicKey) -> bool {
         assert_eq!(
             env::predecessor_account_id(),
             env::current_account_id(),
             "Callback can only be called from the contract"
         );
         let creation_succeeded = is_promise_success();
         if creation_succeeded {
-            Promise::new(env::current_account_id()).delete_key(env::signer_account_pk());
-        } else {
-            // In case of failure, put the amount back.
-            self.accounts
-                .insert(&env::signer_account_pk(), &amount.into());
+            //removing key access to pk
+            self.nft_contract_id = String::from("");
```

Comment:
Why the `nft_contract_id` is overwritten?

Reply:
Fixed: [Commit](https://github.com/Shard-Labs/near-linkdrop/pull/1/commits/ba20ac597220379d4cba8c9ee1401a20e1ab6fda)